doc/md/predicates.md

```diff
@@ -89,6 +89,61 @@ client.Pet.
 
 Custom predicates can be useful if you want to write your own dialect-specific logic or to control the executed queries.
 
+For example, in order to use built-in SQL functions such as `DATE()`, use one of the following options:
+
+1. Pass a dialect-aware predicate function using the `sql.P` option:
+
+```go
+users := client.User.Query().
+	Select("id").
+	Where(sql.P(func(b *sql.Builder) {
+		b.WriteString("DATE(").Ident("last_login_at").WriteByte(')').WriteOp(OpGTE).Arg(value)
+	})).
+	AllX(ctx)
+```
+
+The above code will produce the following SQL query:
+
+```sql
+SELECT `id` FROM `users` WHERE DATE(`last_login_at`) >= ?
+```
+
+2. Inline a predicate expression using the `ExprP()` option:
+
+```go
+users := client.User.Query().
+	Select("id").
+	Where(func(s *sql.Selector) {
+		s.Where(sql.ExprP("DATE(last_login_at >= ?", value))
+	}).
+	AllX(ctx)
+```
+
+But note that `last_login_at` is now hardcoded, unlike the option number 1. 
+
+The above code will produce the same SQL query:
+
+```sql
+SELECT `id` FROM `users` WHERE DATE(`last_login_at`) >= ?
+```
+
+You may also perform a more complex sql query, for example `DATE_ADD()`
+
+```go
+events := client.Event.Query().
+	Where(func(s *sql.Selector) {
+		s.Where(sql.ExprP("DATE_ADD(date, INTERVAL duration MINUTE) BETWEEN ? AND ?", start, end))
+	}).
+	AllX(ctx)
+```
+
+The above code will produce the following SQL query:
+
+```sql
+SELECT `events`.`id`, `events`.`date`, `events`.`duration` 
+FROM events WHERE DATE_ADD(date, INTERVAL duration MINUTE) BETWEEN ? AND ?
+```
+
```

Comment:
```suggestion

```

I feel like the 2 examples above are enough, also this PR sits for a long time and I really want to merge it.